config_src/external/OCEAN_stochastic_phyiscs/MOM_stochastics.F90

```diff
@@ -0,0 +1,157 @@
+  type(verticalGrid_type), intent(in)  :: GV       ! vertical grid structure
+  type(stochastic_CS), pointer,     intent(inout):: CS
+  type(param_file_type),   intent(in)    :: param_file !< A structure to parse for run-time parameters
+  type(diag_ctrl), target, intent(inout) :: diag             !< structure to regulate diagnostic output
+  type(time_type), target                :: Time             !< model time
+  ! Local variables
+  integer,allocatable :: pelist(:) ! list of pes for this instance of the ocean
+  integer :: mom_comm          ! list of pes for this instance of the ocean
+  integer :: num_procs         ! number of processors to pass to stochastic physics
+  integer :: iret              ! return code from stochastic physics
+  integer :: me                !  my pe
+  integer :: master            !  root pe
+  integer :: nx                ! number of x-points including halo
+  integer :: ny                ! number of x-points including halo
+
+! This include declares and sets the variable "version".
+#include "version_variable.h"
+  character(len=40)  :: mdl = "ocean_stochastics_init"  ! This module's name.
+
+  call callTree_enter("ocean_model_stochastic_init(), MOM_stochastics.F90")
+  if (associated(CS)) then
+    call MOM_error(WARNING, "MOM_stochastics_init called with an "// &
+                            "associated control structure.")
+    return
+  else ; allocate(CS) ; endif
+
+  CS%diag => diag
+  CS%Time => Time
+
+  ! Read all relevant parameters and write them to the model log.
+  call log_version(param_file, mdl, version, "")
+
+! get number of processors and PE list for stocasthci physics initialization
+  call get_param(param_file, mdl, "DO_SPPT", CS%do_sppt, &
+                 "If true, then stochastically perturb the thermodynamic "//&
+                 "tendemcies of T,S, amd h.  Amplitude and correlations are "//&
+                 "controlled by the nam_stoch namelist in the UFS model only.", &
+                 default=.false.)
+  call get_param(param_file, mdl, "PERT_EPBL", CS%pert_epbl, &
+                 "If true, then stochastically perturb the kinetic energy "//&
+                 "production and dissipation terms.  Amplitude and correlations are "//&
+                 "controlled by the nam_stoch namelist in the UFS model only.", &
+                 default=.false.)
+  if (CS%do_sppt .OR. CS%pert_epbl) then
+     num_procs=num_PEs()
+     allocate(pelist(num_procs))
+     call Get_PElist(pelist,commID = mom_comm)
+     master=root_PE()
+     nx = grid%ied - grid%isd + 1
+     ny = grid%jed - grid%jsd + 1
+     call init_stochastic_physics_ocn(dt,grid%geoLonT,grid%geoLatT,nx,ny,GV%ke, &
+                                      CS%pert_epbl,CS%do_sppt,master,mom_comm,iret)
+     if (iret/=0)  then
+         write(6,*) 'call to init_stochastic_physics_ocn failed'
+         call MOM_error(FATAL, "stochastic physics in enambled in MOM6 but "// &
```

Comment:
"stochastic physics in enambled in MOM6 but ..." should be "stochastic physics in **enabled** in MOM6 but..."